gp-includes/schema.php

```diff
@@ -145,24 +178,41 @@ function gp_schema_get() {
 	/*
 	 * Meta
 	 */
+
+	/*
+ 	 * The maximum length for the meta_key component of the object_type__meta_key key is limited by the index size limit
+	 * minus the size of the object_type (32 characters).
+	 *
+	 * Also make sure to never go over the length of the column.
+	 */
+	$max_objtype_metakey_key_length = min( $max_index_length - 32, 255 );
+
+	/*
+ 	 * The maximum length for the meta_key component of the object_type__object_id__meta_key key is limited by the index size limit
+	 * minus the size of the object_type (32 characters) and the object_id (8 bytes = 2 characters).
+	 *
+	 * Also make sure to never go over the length of the column.
+	 */
+	$max_objtype_objid_metakey_key_length = min( $max_index_length - 32 - 2, 255 );
+
 	$gp_schema['meta'] = "CREATE TABLE $wpdb->gp_meta (
 		meta_id bigint(20) NOT NULL auto_increment,
 		object_type varchar(32) NOT NULL default 'gp_option',
 		object_id bigint(20) NOT NULL default 0,
-		meta_key varchar($max_index_length) DEFAULT NULL,
+		meta_key varchar(255) DEFAULT NULL,
 		meta_value longtext DEFAULT NULL,
 		PRIMARY KEY  (meta_id),
-		KEY object_type__meta_key (object_type,meta_key),
-		KEY object_type__object_id__meta_key (object_type,object_id,meta_key)
+		KEY object_type__meta_key (object_type,meta_key({$max_objtype_metakey_key_length})),
+		KEY object_type__object_id__meta_key (object_type,object_id,meta_key({$max_objtype_objid_metakey_key_length}))
 	) $charset_collate;";
 
 	/*
 	 * Permissions
 	 */
 	$gp_schema['permissions'] = "CREATE TABLE $wpdb->gp_permissions (
 		id int(10) NOT NULL AUTO_INCREMENT,
-		user_id int(10) DEFAULT NULL,
-		action varchar(255) DEFAULT NULL,
+		user_id bigint(20) DEFAULT NULL,
+		action varchar(20) DEFAULT NULL,
```

Comment:
I chose 20 as we use that length for the status field, but I'm not tied to it by any means.

I looked through our current actions at I believe the longest one was 9.  255 seemed excessive.  We could go somewhere in between 64 or 128 would both be reasonable sizes and not cause an index issue.